### DISCUSSION TITLE
Help with extracting network key while binding Aqara TVOC to M2 hub

### DISCUSSION THREAD

**Discussion by ptr727** [Koenkk/zigbee2mqtt#13976] [Devices]
I have some time and equipment on hand, and I am trying to create a PR for https://github.com/Koenkk/zigbee2mqtt/issues/13464  
I.e. I am trying to figure out how the Aqara app sets the units for VOC and temperature, such that I can do it on Z2M, instead of having to use the Aqara Hub app, and then rebind to Z2M.

I tried following instructions from the [Z2M docs](https://www.zigbee2mqtt.io/advanced/zigbee/04_sniff_zigbee_traffic.html#_3-sniffing-traffic), and from a [blog](https://www.hal9k.dk/sniffing-philips-hue-zigbee-traffic-with-wireshark/) post, and from various YT videos and google search images.

Main problem is that either I am not capturing the key exchange, or the current tool versions do not look like the images in instructions.

I tried using the standard TI [sniffer firmware](https://www.ti.com/tool/PACKET-SNIFFER) with a trial version of the [Ubiqua Protocol Analyzer](https://www.ubilogix.com/ubiqua/).  
And I tried using the [ZBOSS firmware](https://dsr-iot.com/downloads) with WireShark.

At no point do I see "Device Announcement" or "Transport Key" messages as depicted in the instruction resources.

I am wondering if I am doing something wrong, or if the current versions of the tools work differently?  
I have captured packets while joining, happy to share in private if somebody can point me in the right direction?

**Comment by kirovilya** [comment]
A little trick for sniffing new devices from Aqara - before pairing a new Aqara device, pair some old sensor (temperature sensor, contact, button and other). In the process of pairing the old sensor, a key will be issued that will allow you to decrypt the events of the new device :)

**Reply by ptr727** [reply]
Ok, I'll give that a try.  
Can you elaborate a bit on why I'd see a key exchange for a non-aqara being paired with an aqara hub? (I was wondering if they are pre-keyed?)

**Reply by ptr727** [reply]
Ok, I tried to pair a linkind magnet with the M2 hub, never got to capture any key exchanges.
I paired a TVOC sensor with Z2M, got `Transport Key`,  `Request Key`, `Verify Key`, and `Confirm Key` messages.
Then re-paired the same TVOC with the M2, it pairs successfully, but I see no key exchanges.

Any other ideas?

**Reply by kirovilya** [reply]
> Any other ideas?

try another device instead of the contact sensor.

New devices have a fixed key, which is requested by the hub from the Aqara cloud during pairing. So it can work with their hubs without key exchange.

But this only affects the pairing process - the TC network key is then used - this is what you need to get when pairing the old device.

Do the pairing of both devices in one sniff process - then you will immediately get the information

**Reply by ptr727** [reply]
Got it, I paired an older temp/hum sensor, and got the key.
Pretty interesting that they bind with no key exchange, is it for convenience or some security feature?

**Reply by kirovilya** [reply]
security feature
